src/Agent.Worker/TestResults/Telemetry/TelemetryPublisher.cs

```diff
@@ -0,0 +1,44 @@
+﻿using Microsoft.VisualStudio.Services.CustomerIntelligence.WebApi;
+using Microsoft.VisualStudio.Services.WebApi;
+using Microsoft.VisualStudio.Services.WebPlatform;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+
+namespace Microsoft.VisualStudio.Services.Agent.Worker.TestResults
+{
+    public class TelemetryPublisher
+    {
+        private static TelemetryPublisher publisher;
+        private static readonly object _cacheLock = new object();
+        private static CustomerIntelligenceHttpClient _ciClient;
+
+        private TelemetryPublisher(VssConnection connection)
+        {
+            _ciClient = connection.GetClient<CustomerIntelligenceHttpClient>();
+        }
+
+        public static TelemetryPublisher GetInstance(VssConnection connection)
```

Comment:
Since we ask for connection here and if the connection is different. You will be using the old connection object.
In our case, it may not happen. but we don't know.

And Agent.Worker dies every time the job is run. So inside a job if there are multiple PTR tasks, it's okay to create get the client that many times I think. I don't see a real benefit here ?